### PR TITLE
Dicom multiple audiences

### PR DESCRIPTION
**Pull request by QuanWanxx**
Support get DICOM server token in different cloud environment groups.

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Common/Configurations/DicomServerConfiguration.cs [hunk 1]
```diff
@@ -17,5 +17,8 @@ public class DicomServerConfiguration
 
         [JsonProperty("authentication")]
         public AuthenticationType Authentication { get; set; } = AuthenticationType.None;
+
+        [JsonProperty("serverEnvironmentGroup")]
+        public ServerEnvironmentGroup ServerEnvironmentGroup { get; set; } = ServerEnvironmentGroup.PROD;
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Common/Configurations/DicomServerConfiguration.cs:22]: Maybe we can directly use Audience directly here?

File: FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.DataClient/Api/Dicom/DicomApiConstants.cs [hunk 2]
```diff
@@ -47,8 +47,13 @@ public class DicomApiConstants
...
         /// <summary>
-        /// The managed identity url to get access token for DICOM service.
+        /// The managed identity audiences to get access token for DICOM service.
         /// </summary>
-        public const string DicomResourceUrl = "https://dicom.healthcareapis.azure.com";
+        public static readonly Dictionary<ServerEnvironmentGroup, string> DicomServerAudienceMap = new Dictionary<ServerEnvironmentGroup, string>()
+        {
+            { ServerEnvironmentGroup.PROD, "https://dicom.healthcareapis.azure.com" },
+            { ServerEnvironmentGroup.TEST, "https://dicom.healthcareapis.azure-test.net" },
+            { ServerEnvironmentGroup.LOCAL, "https://dicom.healthcareapis.azure-test.net" },
```
Comment (BoyaWu10) [inline FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.DataClient/Api/Dicom/DicomApiConstants.cs:56]: Are the values for TEST and LOCAL environment the same?

Comment (BoyaWu10) [reply FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.DataClient/Api/Dicom/DicomApiConstants.cs:56]: If so, maybe we can just keep TEST here?

Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.DataClient/Api/Dicom/DicomApiConstants.cs]: Remove this property and directly set up audience in appsetting configuration.

File: FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Common/Configurations/DicomServerConfiguration.cs (hunk 1 above)
Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Common/Configurations/DicomServerConfiguration.cs]: Sure, updated.

File: FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Common/Configurations/DicomServerConfiguration.cs [hunk 3]
```diff
@@ -17,5 +17,10 @@ public class DicomServerConfiguration
 
         [JsonProperty("authentication")]
         public AuthenticationType Authentication { get; set; } = AuthenticationType.None;
+
+        // Azure DICOM server default audience is "https://dicom.healthcareapis.azure.com".
+        // Refer https://learn.microsoft.com/en-us/azure/healthcare-apis/dicom/dicom-get-access-token-azure-cli-old#obtain-a-token
+        [JsonProperty("audience")]
+        public string Audience { get; set; } = "https://dicom.healthcareapis.azure.com";
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Common/Configurations/DicomServerConfiguration.cs:24]: need we provide the audience field in appsetting.json?

Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Common/Configurations/DicomServerConfiguration.cs:24]: const string for this url?

Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Common/Configurations/DicomServerConfiguration.cs]: Moved to **ConfigurationConstants**

Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Common/Configurations/DicomServerConfiguration.cs]: Added audience in **appsettings-example-v2.json**

File: FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Tool/Microsoft.Health.AnalyticsConnector.Tool.csproj [hunk 4]
```diff
@@ -25,9 +25,6 @@
   </ItemGroup>
 
   <ItemGroup>
-    <None Update="appsettings-example-v1.json">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </None>
     <None Update="appsettings.json">
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Tool/Microsoft.Health.AnalyticsConnector.Tool.csproj:28]: do we need to copy the `appsettings-example-v2.json` to the output?

File: FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Tool/appsettings-example-v2.json [hunk 5]
```diff
@@ -5,7 +5,8 @@
     "dicomServer": {
       "serverUrl": "",
       "apiVersion": "v1",
-      "authentication": "None"
+      "authentication": "None",
+      "audience": "https://dicom.healthcareapis.azure.com"
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Tool/appsettings-example-v2.json:9]: add it to E2ETests project too?

File: FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Tool/appsettings-example-v2.json [hunk 6]
```diff
@@ -5,7 +5,8 @@
     "dicomServer": {
       "serverUrl": "",
       "apiVersion": "v1",
-      "authentication": "None"
+      "authentication": "None",
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Tool/appsettings-example-v2.json:8]: just curious, what is the current configuration version? if v2, maybe we need to update the default appsetting.json in E2Etest and functionApp to v2.

**Review by moria97 — APPROVED** [review]

File: FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Tool/Microsoft.Health.AnalyticsConnector.Tool.csproj (hunk 4 above)
Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Tool/Microsoft.Health.AnalyticsConnector.Tool.csproj:28]: No, this is just an example for user on how to write the ConfigVersion 2 appsettings.

File: FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Tool/appsettings-example-v2.json (hunk 6 above)
Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Tool/appsettings-example-v2.json]: v2 is for DICOM pipeline (Even though V2 also support FHIR data source), the PaaS Analytics Connector now still uses v1 when deploying FHIR pipeline. So I think we can still use v1 in our E2E test.

And our DICOM E2E test all use v2 appsetting.

We seems haven't update configVersion for FunctionApp.

File: FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Tool/appsettings-example-v2.json (hunk 5 above)
Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Tool/appsettings-example-v2.json:9]: Added, thanks.

**Pull request merged**